lib/std/os.zig

```diff
@@ -254,6 +254,101 @@ pub fn close(fd: fd_t) void {
     }
 }
 
+pub const FChmodError = error{
+    AccessDenied,
+    InvalidFileDescriptor,
+    InputOutput,
+    SymLinkLoop,
+    FileNotFound,
+    SystemResources,
+    ReadOnlyFileSystem,
+} || UnexpectedError;
+
+/// Changes the mode of the file referred to by the file descriptor.
+/// The process must have the correct privileges in order to do this
+/// successfully, or must have the effective user ID matching the owner
+/// of the file.
+pub fn fchmod(fd: fd_t, mode: mode_t) FChmodError!void {
+    while (true) {
+        const res = if (builtin.link_libc) blk: {
+            const rc = std.c.fchmod(fd, mode);
```

Comment:
Can't this and the cases below be folded into a single one reaching into `system`?

Reply:
Oh yep, I can do that now.